Updates CustomFilterInput extension point

vinamartin opened the pull request:
Updates CustomFilterInput to take in **value** and **onChange** to be more usable across the UI.
Previously, you wouldn't be able to use this CustomFilterInput in the various fields of "Near" so you could only use this for single input filter types.

To test, create a CustomFilterInput in a downstream project, for example here, I've just made a simple textfield with the blue coloring on the outside. Verify you can see the custom filter input for only string property queries (title, anytext, etc) and that contains shows your custom filter input. Verify IS EMPTY doesn't show your custom filter input. Verify NEAR shows your custom filter input for both fields. 
![Screen Shot 2020-12-15 at 10 54 03](https://user-images.githubusercontent.com/9013407/102253077-2201fd80-3ec4-11eb-90b3-4623a814c36f.png)
![Screen Shot 2020-12-15 at 10 54 13](https://user-images.githubusercontent.com/9013407/102253098-26c6b180-3ec4-11eb-9386-9c20409a040f.png)

Commit:
Refactors customFilterInput to be more usable

andrewkfiedler commented on ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/near.tsx:
```diff
@@ -23,6 +24,36 @@ type NearFieldProps = {
   onChange: (val: ValueTypes['proximity']) => void
 }
 
+const showCustomTextFieldOrDefault = ({
```
Is it possible to factor this out into a common component between near and the filterInput?

vinamartin replied:
Yeah, I think so since they seem to be more or less the same thing 👍

Commit:
Creates common component for textfields

jordanwilking commented on ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/filter/filter-input/customInputOrDefault.tsx:
```diff
@@ -0,0 +1,39 @@
+/**
```
I asked about the copyright because I wanted to get rid of these ugly things 😢

Commit:
Fix ts issues

Commit:
Fixes bugs:
1. Fixes bug where switching between near and contains broke the ui
2. Fixes issue where input wouldn't work for some interactions

jordanwilking commented:
📶 Hero Successful 📶 

#### Steps

- Used custom component downstream (detailed steps can be found there)

stevenmalmgren commented on ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/filter/filter-input/customInputOrDefault.tsx:
```diff
@@ -0,0 +1,52 @@
+  onChange: (e: string) => void
+  props?: Partial<TextFieldProps>
+}) => {
+  let textValue = value
+  //Clear out value when switching between structured string inputs (e.g. NEAR)
+  if (typeof textValue !== 'string') {
+    textValue = ''
+  }
```
```suggestion
  const textValue = typeof value !== string ?  ' ' : value
```
